frontend/app/controllers/spree/checkout_controller.rb

```diff
@@ -76,106 +77,135 @@ def ensure_valid_state
+    end
 
-      def setup_for_current_state
-        method_name = :"before_#{@order.state}"
-        send(method_name) if respond_to?(method_name, true)
-      end
+    def setup_for_current_state
+      method_name = :"before_#{@order.state}"
+      send(method_name) if respond_to?(method_name, true)
+    end
 
-      def before_address
-        # if the user has a default address, a callback takes care of setting
-        # that; but if he doesn't, we need to build an empty one here
-        @order.bill_address ||= Address.build_default
-        @order.ship_address ||= Address.build_default if @order.checkout_steps.include?('delivery')
-      end
+    def before_address
+      # if the user has a default address, a callback takes care of setting
+      # that; but if he doesn't, we need to build an empty one here
+      @order.bill_address ||= Address.build_default
+      @order.ship_address ||= Address.build_default if @order.checkout_steps.include?('delivery')
+    end
 
-      def before_delivery
-        return if params[:order].present?
+    def before_delivery
+      return if params[:order].present?
 
-        packages = @order.shipments.map { |s| s.to_package }
+      packages = @order.shipments.map(&:to_package)
+      @differentiator = Spree::Stock::Differentiator.new(@order, packages)
+    end
+
+    def before_payment
+      if @order.checkout_steps.include? "delivery"
+        packages = @order.shipments.map(&:to_package)
         @differentiator = Spree::Stock::Differentiator.new(@order, packages)
+        @differentiator.missing.each do |variant, quantity|
+          @order.contents.remove(variant, quantity)
+        end
+
+        # @order.contents.remove did transitively call reload in the past.
+        # Hiding the fact that the machine advanced already to "payment" state.
+        #
+        # As an intermediary step to optimize reloads out of high volume code path
+        # the reload was lifted here and will be removed by later passes.
+        @order.reload
       end
 
-      def before_payment
-        if @order.checkout_steps.include? "delivery"
-          packages = @order.shipments.map { |s| s.to_package }
-          @differentiator = Spree::Stock::Differentiator.new(@order, packages)
-          @differentiator.missing.each do |variant, quantity|
-            @order.contents.remove(variant, quantity)
-          end
+      if try_spree_current_user && try_spree_current_user.respond_to?(:payment_sources)
+        @payment_sources = try_spree_current_user.payment_sources
+      end
+    end
 
-          # @order.contents.remove did transitively call reload in the past.
-          # Hiding the fact that the machine advanced already to "payment" state.
-          #
-          # As an intermediary step to optimize reloads out of high volume code path
-          # the reload was lifted here and will be removed by later passes.
-          @order.reload
-        end
+    def add_store_credit_payments
```

Comment:
Assignment Branch Condition size for add_store_credit_payments is too high. [16.28/15]